datalad/distribution/dataset.py

```diff
@@ -102,14 +103,44 @@ def repo(self):
                             pass
                 if self._repo is None:
                     lgr.info("Failed to detect a valid repo at %s" % self.path)
+                else:
+                    # reset ID, because we now have a repo attached
+                    self._id = None
         elif not isinstance(self._repo, AnnexRepo):
             # repo was initially set to be self._repo but might become AnnexRepo
             # at a later moment, so check if it didn't happen
             if 'git-annex' in self._repo.get_branches():
                 # we acquired git-annex branch
                 self._repo = AnnexRepo(self._repo.path, create=False)
+                # reset ID, because we now have an annex attached
+                self._id = None
```

Comment:
Rational: we already have dataset.path that is unique per dataset on a
system and doesn't change over the lifetime of the object. No use
duplicating just this behavior with yet another ID. This new implementation
adds the longevity. Whenever possible it yields an ID that can be traced
back to from future versions and other machines.

On Aug 25, 2016 00:30, "Yaroslav Halchenko" <notifications@github.com>
wrote:

> In datalad/distribution/dataset.py
> <https://github.com/datalad/datalad/pull/720#discussion_r76151647>:
>
> >          elif not isinstance(self._repo, AnnexRepo):
> >              # repo was initially set to be self._repo but might become AnnexRepo
> >              # at a later moment, so check if it didn't happen
> >              if 'git-annex' in self._repo.get_branches():
> >                  # we acquired git-annex branch
> >                  self._repo = AnnexRepo(self._repo.path, create=False)
> > +                # reset ID, because we now have an annex attached
> > +                self._id = None
>
> So it would possibly change in the lifetime of a dataset... What is the
> purpose of id here since we wouldn't then rely on id to correspond to the
> dataset? ( Since it might change)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/datalad/datalad/pull/720/files/9b0f4a934c535119d7146704bab253cdc671c86c#r76151647>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAIVHzmLdM729N9WkUbsPi_t697UOpyWks5qjMYEgaJpZM4JsSSH>
> .
>